entities/models.py

```diff
@@ -0,0 +1,99 @@
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+from MyAcademy.settings import MEDIA_URL
+
+
+class Entity(models.Model):
+	verified = models.BooleanField(default=False)
+	picture = models.ImageField(upload_to='profile_pictures', blank=True)
+
+	@property
+	def get_picture(self):
+		if (self.picture):
+			return self.picture.url
+		else:
+			return MEDIA_URL + self._meta.get_field('picture').default
+
+	class Meta:
+		abstract = True
+
+
+class Professor(Entity):
+	INSTRUCTOR = 'ins'
+	ASSISTANT = 'asi'
+	ASSOCIATE = 'aso'
+	FULL = 'ful'
+	ACADEMIC_RANK = [
+		(INSTRUCTOR, 'مدرس'),
+		(ASSISTANT, 'استادیار'),
+		(ASSOCIATE, 'دانشیار'),
+		(FULL, 'استاد'),
+	]
+
+	first_name = models.CharField(_("first name"), max_length=30, blank=False)
+	last_name = models.CharField(_("last name"), max_length=30, blank=False)
+	my_department = models.ForeignKey('Department', verbose_name="نام دانشکده", on_delete=models.SET_NULL, null=True)
+	rank = models.CharField(max_length=3, choices=ACADEMIC_RANK, verbose_name="مرتبه‌ی علمی", default=INSTRUCTOR)
+
+	@property
+	def name(self):
+		return self.first_name + " " + self.last_name
+
+	def __init__(self, *args, **kwargs):
+		super().__init__(*args, **kwargs)
+		self._meta.get_field('picture').default ='/profile_pictures/professor_default_picture.png'
+
+	def __str__(self):
+		return self.name
+
+	class Meta:
+		verbose_name = _("استاد")
+		verbose_name_plural = _("اساتید")
+		constraints = [
+			models.UniqueConstraint(fields=['first_name', 'last_name', 'my_department'], name='unique_professor'),
+		]
+
+
+class Department(Entity):
+	name = models.CharField(_("name"), max_length=30, blank=False)
+	my_university = models.ForeignKey('University', verbose_name="دانشکده", on_delete=models.SET_NULL, null=True)
```

Comment:
This should be دانشگاه maybe.
Also, I think that these names (my university, my department) could be truncated to university and department.

Reply:
Yes, the ```verbose_name``` of ```my_university``` is definitely دانشگاه.
For changing ```my_university``` and ```my_department``` to ```university``` and ```department``` I'm a bit skeptical because they have to be manually changed in a lot of places. @amsabour could you make the changes onegaishimasu? (especially the first one)

Reply:
The first one is now fixed. 

As for the second one, I remember them being `university` and `department` in the beginning but django was freaking out because it apparently can't be the same name as the model class. So that's why these names were used.